controllers/common.go

```diff
@@ -25,3 +61,55 @@ func ReconcileBatch(l logr.Logger, reconcileFuncs ...ReconcileFunc) (bool, error
 	}
 	return true, nil
 }
+
+func PopulateResticSecret(name string, namespace string, label string) (*corev1.Secret, error) {
+
+
+	// define Restic secret to be created
+	newResticSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-secret", name),
+			Namespace: namespace,
+			Labels: map[string]string{
+				label: name,
+			},
+		},
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	return newResticSecret, nil
+}
+
+func BuildResticSecret(givensecret *corev1.Secret, secret *corev1.Secret, resticrepo string) error {
+
+	if resticrepo == "" {
+		return errors.New("restic repo value is empty")
+	}
+	// assign new restic secret values
+	for key, val := range givensecret.Data {
```

Comment:
Are we ok if the values `AWSAccessKey` , `AWSSecretKey` ,`AWSDefaultRegion` etc are empty ?

Reply:
so far we are validating anywhere. I can add a validate func..